### PR TITLE
G3_MASTER: Injector module fixes

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules_dart/transform/lib/src/transform/directive_metadata_linker/ng_meta_linker.dart [hunk 1]
```diff
@@ -193,7 +193,11 @@ class _NgMetaIdentifierResolver {
       } else if (meta is CompilePipeMetadata) {
         _resolveDiDependencyMetadata(ngMetaMap, meta.type.name, meta.type.diDeps);
       } else if (meta is CompileInjectorModuleMetadata) {
-        _resolveDiDependencyMetadata(ngMetaMap, meta.name, meta.diDeps);
+        if (meta.injectable) {
+          // Only resolve constructor arguments if the InjectorModule is marked as
+          // @Injectable.
```
Comment (vicb) [inline modules_dart/transform/lib/src/transform/directive_metadata_linker/ng_meta_linker.dart]: is this true (`@Component`, `@Directive`, `@Pipe`)

File: modules_dart/transform/test/transform/directive_metadata_linker/all_tests.dart [hunk 2]
```diff
@@ -767,29 +767,66 @@ void allTests() {
       expect(inj.providers[1].useClass.identifier.moduleUrl, equals("moduleUrl"));
     });
 
-    test('should resolve constructor deps', () async {
+    test('should resolve constructor deps if marked as @Injectable', () async {
```
Comment (vicb) [inline modules_dart/transform/test/transform/directive_metadata_linker/all_tests.dart]: same comment

File: modules_dart/transform/test/transform/directive_metadata_linker/all_tests.dart [hunk 3]
```diff
@@ -767,29 +767,66 @@ void allTests() {
...
 
       expect(inj.diDeps[0].token.identifier.name, equals("Service"));
       expect(inj.diDeps[0].token.identifier.moduleUrl, equals("moduleUrl"));
 
+      expect(inj.diDeps[1].token.identifier.name, equals("UnknownId"));
+      expect(inj.diDeps[1].token.identifier.moduleUrl, isNull);
+
+    });
+
+    test('should not resolve constructor deps if not marked as @Injectable', () async {
```
Comment (vicb) [inline modules_dart/transform/test/transform/directive_metadata_linker/all_tests.dart]: same

File: modules/angular2/test/core/di/injector_spec.ts [hunk 4]
```diff
@@ -15,32 +14,10 @@ export function main() {
        () => { expect(Injector.NULL.get('someToken', 'notFound')).toEqual('notFound'); });
   });
 
-  describe('MapInjector', () => {
-    it('should throw if not found', () => {
-      expect(() => new MapInjector(null, new Map<any, any>()).get('someToken'))
-          .toThrowError('No provider for someToken!');
-    });
+  describe('InjectorFactory.Empty', () => {
```
Comment (vicb) [inline modules/angular2/test/core/di/injector_spec.ts]: .EMPTY

**Comment by vicb** [comment]
LGTM when CI green (one minor comment)

**Comment by tbosch** [comment]
Landed in g3_master

**Pull request closed without merging**